src/components/WeatherCard.svelte

```diff
@@ -1,75 +1,65 @@
 <script lang="ts">
-  export let data: any; // Weather data
-  export let wmoCodeDescription: String;
-  export let temperatureUnit: String = "°C"
-  let widgets = [1, 2, 3]; // Widgets for the top row
-  let largeWidget = 4; // Large widget for the bottom
+    import CurrentWeatherConditionWidget from "./widgets/CurrentWeatherConditionWidget.svelte";
+    import DailyWeatherConditionWidget from "./widgets/DailyWeatherConditionWidget.svelte";
+    export let data: any; // Weather data
+    export let wmoCodeDescription: String;
+    export let temperatureUnit: String = "°C";
 </script>
 
 <div class="card">
-  <div class="grid">
-    <div class="widget">
-      <h1>
-        {Object.keys(data).length === 0
-        ? ""
-        : JSON.stringify(data.current.temperature_2m) + temperatureUnit}
-      </h1>
-      <br>
-      {wmoCodeDescription}  
+    <div class="grid">
+        <CurrentWeatherConditionWidget
+            {data}
```

Comment:
Should we maybe consider using [Svelte stores](https://svelte.dev/docs/svelte-store) for handling stateful data instead of passing it into the functions? This doesn't have to be done in this PR but a more general technical improvement that we could make.

Reply:
That's a good point. I just looked into it an it seems in can also help in creating a centralized state management solution which is a good thing. I'll create a new issue for that.